### PR TITLE
S2t retrieve and display ingredients

### PR DESCRIPTION
**Pull request by DanielGardiner**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,4097 @@
+{
```
Comment (jasonrees1) [inline package-lock.json]: Mike: "why have you added all this shit, this means that they havent merged master when I told them to"

I think that he wants you to delete this file as its now in masters .gitignore

File: src/Ingredients/DisplayIngredients.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+
+namespace BreadAndIfit\Ingredients;
+
+
+class DisplayIngredients
+{
+    public static function displayIngredients(array $ingredients): string {
```
Comment (jasonrees1) [inline src/Ingredients/DisplayIngredients.php]: should this { be on the next line?

File: src/Ingredients/Ingredient.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+namespace BreadAndIfit\Ingredients;
+
+
+class Ingredient
+{
+    protected $id;
+    protected $name;
+    protected $category;
+
+    public function getId(): int {
```
Comment (jasonrees1) [inline src/Ingredients/Ingredient.php]: psr2

File: tests/Ingredients/DisplayIngredientsTest.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+
+use BreadAndIfit\Ingredients\DisplayIngredients;
+use BreadAndIfit\Ingredients\Ingredient;
+use PHPUnit\Framework\TestCase;
+
+require('../../src/Ingredients/DisplayIngredients.php');
+require('../../src/Ingredients/Ingredient.php');
+
+$carrot = new class() extends Ingredient {
```
Comment (jasonrees1) [inline tests/Ingredients/DisplayIngredientsTest.php]: should this be class and if so should it have a capital letter?

File: tests/Ingredients/DisplayIngredientsTest.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+
+use BreadAndIfit\Ingredients\DisplayIngredients;
+use BreadAndIfit\Ingredients\Ingredient;
+use PHPUnit\Framework\TestCase;
+
+require('../../src/Ingredients/DisplayIngredients.php');
+require('../../src/Ingredients/Ingredient.php');
```
Comment (jasonrees1) [inline tests/Ingredients/DisplayIngredientsTest.php]: unit test should be run in the test folder

File: tests/Ingredients/DisplayIngredientsTest.php (hunk 4 above)
Comment (jasonrees1) [reply tests/Ingredients/DisplayIngredientsTest.php]: see below also

File: tests/Ingredients/DisplayIngredientsTest.php (hunk 5 above)
Comment (jasonrees1) [reply tests/Ingredients/DisplayIngredientsTest.php]: Though your test works fine otherwise

**Review by jasonrees1 — CHANGES_REQUESTED** [review]

File: src/Ingredients/DisplayIngredients.php [hunk 6]
```diff
@@ -16,6 +16,13 @@ public static function displayIngredients(array $ingredients): string {
 
     }
 
+    /**
+     * This method formats the array of data into a multidimensional associative array of ingredients
+     * organised by their category.
+     * @param array $ingredients is the array of objects that came from the database.
+     * @return array
```
Comment (avocadobelly) [inline src/Ingredients/DisplayIngredients.php]: please elaborate on array

File: src/Ingredients/DisplayIngredients.php [hunk 7]
```diff
@@ -24,6 +31,12 @@ private static function formatData(array $ingredients): array {
         return $ingredientSorted;
     }
 
+    /**
```
Comment (avocadobelly) [inline src/Ingredients/DisplayIngredients.php:37]: Add description for what the whole method does

File: tests/Ingredients/DisplayIngredientsTest.php [hunk 8]
```diff
@@ -5,36 +5,30 @@
...
-            protected $category = 'Vegetable';
-        };
+        $ingredientMock = $this->createMock(Ingredient::class);
+        $ingredientMock->expects($this->once())->method('getCategory')->willReturn('Vegetables');
+        $ingredientMock->expects($this->once())->method('getName')->willReturn('carrot');
+
+        $result = DisplayIngredients::displayIngredients([$ingredientMock]);
 
-        $result = DisplayIngredients::displayIngredients(['Vegetable' => $carrot]);
+        var_dump($result);
```
Comment (avocadobelly) [inline tests/Ingredients/DisplayIngredientsTest.php]: is this the best way to display this information, using vardump?

**Pull request merged**